libraries/AP_HAL_Linux/Storage.cpp

```diff
@@ -11,6 +11,17 @@
 #include <AP_HAL/AP_HAL.h>
 #include <AP_Vehicle/AP_Vehicle_Type.h>
 
+// Necessary for toolchains that does not provide `strndupa`, such as musl.
+#ifndef __GLIBC__
+static inline char* strndupa(const char *old_string, const size_t size)
+{
+    const size_t string_size = strnlen(old_string, size);
+    char *new_string = static_cast<char*>(alloca(string_size + 1));
```

Comment:
is this valid? alloca() context disappears on call return. Is the inline sufficient to guarantee we don't pop the stack frame?
I'd prefer to remove the single call to strndupa()

Reply:
iff it remains inline... which means either giving it the always_inline attribute or implementing it as a macro (what I actually did in kmod: https://git.kernel.org/pub/scm/utils/kernel/kmod/kmod.git/tree/shared/missing.h#n37

Removing the only user is also a good option

Reply:
Moved to missing folder using the "statement exprs" extension, as pointed in the example.

I'm more inclined to add it in "missing" since this function may be used in other files,
replace this function may result in duplicated code in the future, besides that is much simpler to
know that "strndupa" does.